app/code/community/Oyst/OneClick/Model/Payment/Data.php

```diff
@@ -423,10 +425,8 @@ protected function _constructParams()
     protected function _addAmount(&$params)
     {
         $checkoutSession = Mage::getSingleton('checkout/session');
-        $params['amount'] = array(
-            'value' => $checkoutSession->getQuote()->getGrandTotal() * 100,
-            'currency' => $checkoutSession->getQuote()->getQuoteCurrencyCode(),
-        );
+        $amount = new OystPrice($checkoutSession->getQuote()->getGrandTotal() * 100, 'EUR');
```

Comment:
the ```* 100``` have to be removed because the lib manage already this.
https://github.com/oystparis/oyst-php/blob/master/src/Classes/OystPrice.php#L54